tests/grammar_test.go

```diff
@@ -334,77 +334,75 @@ func TestUnevenNumberOfDigits(t *testing.T) {
 	}
 }
 
-func TestNonAsciiCharacters(t *testing.T) {
-	// Non-ascii characters are NOT accepted in text strings.
+func TestInvalidCharacters(t *testing.T) {
+	// Non-UTF8 characters are NOT accepted in text strings.
 	_, err := gyp.ParseString(`
-	rule NON_ASCII {
+	rule TEST {
 		strings:
 			$s1 = "foo` + "\xe8" + `bar"
 		condition:
 			all of them
 	}`)
 	if assert.Error(t, err) {
-		assert.Equal(t, `line 4: non-ascii character "\xe8"`, err.Error())
+		assert.Equal(t, `line 4: invalid UTF-8 character "\xe8"`, err.Error())
 	}
 
-	// Non-ascii characters are NOT accepted in regexps.
+	// Non-UTF8 characters are NOT accepted in regexps.
 	_, err = gyp.ParseString(`
-	rule NON_ASCII {
+	rule TEST {
 		strings:
-			$s1 = /foo` + "\x03" + `bar/
+			$s1 = /foo` + "\xC3" + `bar/
 		condition:
 			all of them
 	}`)
 	if assert.Error(t, err) {
-		assert.Equal(t, `line 4: non-ascii character "\x03"`, err.Error())
+		assert.Equal(t, `line 4: invalid UTF-8 character "\xc3"`, err.Error())
 	}
 
 	// Non-ascii characters are NOT accepted in argument to base64 modifier.
 	_, err = gyp.ParseString(`
-	rule NON_ASCII {
+	rule TEST {
 		strings:
 			$s1 = "foo" base64("é")
 		condition:
 			all of them
 	}`)
 	if assert.Error(t, err) {
-		assert.Equal(t, `line 4: non-ascii character "\xc3"`, err.Error())
+		assert.Equal(t, `line 4: invalid ASCII character "\xc3"`, err.Error())
 	}
 
-	// Non-ascii characters are NOT accepted in string literals.
+	// UTF8 characters are accepted in literal strings.
```

Comment:
We should probably assert that UTF8 chars are accepted in defined strings and regexps too, or cover those cases in `testRules`.

Reply:
Done